src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordClone.cs

```diff
@@ -7,118 +7,115 @@
-
-        internal override bool HasDeclarativeSecurity => false;
-
-        internal override MarshalPseudoCustomAttributeData? ReturnValueMarshallingInformation => null;
-
-        internal override bool RequiresSecurityObject => false;
-
-        internal override CallingConvention CallingConvention => CallingConvention.HasThis;
-
-        internal override bool GenerateDebugInfo => false;
+            if (ContainingType.IsAbstract)
+            {
+                result &= ~DeclarationModifiers.Virtual;
+                result |= DeclarationModifiers.Abstract;
+            }
 
-        public override DllImportData? GetDllImportData() => null;
+            Debug.Assert((result & ~allowedModifiers) == 0);
+#if DEBUG
+            Debug.Assert(modifiersAreValid(result));
+#endif 
+            return result;
 
-        internal override ImmutableArray<string> GetAppliedConditionalSymbols()
-            => ImmutableArray<string>.Empty;
+#if DEBUG
+            static bool modifiersAreValid(DeclarationModifiers modifiers)
+            {
+                if ((modifiers & DeclarationModifiers.AccessibilityMask) != DeclarationModifiers.Public)
+                {
+                    return false;
+                }
 
-        internal override IEnumerable<SecurityAttribute> GetSecurityInformation()
-            => Array.Empty<SecurityAttribute>();
+                modifiers &= ~DeclarationModifiers.AccessibilityMask;
 
-        internal override bool IsMetadataNewSlot(bool ignoreInterfaceImplementationChanges = false) => !IsOverride;
+                switch (modifiers)
+                {
+                    case DeclarationModifiers.None:
+                        return true;
+                    case DeclarationModifiers.Abstract:
+                        return true;
+                    case DeclarationModifiers.Override:
+                        return true;
+                    case DeclarationModifiers.Abstract | DeclarationModifiers.Override:
+                        return true;
+                    case DeclarationModifiers.Virtual:
+                        return true;
+                    default:
+                        return false;
+                }
+            }
+#endif 
+        }
 
-        internal override bool IsMetadataVirtual(bool ignoreInterfaceImplementationChanges = false) => true;
+        protected override (TypeWithAnnotations ReturnType, ImmutableArray<ParameterSymbol> Parameters, bool IsVararg, ImmutableArray<TypeParameterConstraintClause> DeclaredConstraintsForOverrideOrImplementation) MakeParametersAndBindReturnType(DiagnosticBag diagnostics)
+        {
+            HashSet<DiagnosticInfo>? ignoredUseSiteDiagnostics = null; // This is reported when we bind bases
+            NamedTypeSymbol baseType = ContainingType.BaseTypeNoUseSiteDiagnostics;
+            return (ReturnType: !baseType.IsObjectType() && FindValidCloneMethod(baseType, ref ignoredUseSiteDiagnostics) is { } baseClone ?
```

Comment:
>!baseType.IsObjectType() && FindValidCloneMethod(baseType, ref ignoredUseSiteDiagnostics) is { } baseClone [](start = 32, length = 106)

Consider extracting a helper method for use here and in `MakeDeclarationModifiers()`. It looks like the helper method would also encapsulate `ignoredUseSiteDiagnostics` and `baseType` locals.